Add `CupertinoPicker` interactive example

TahaTesser opened the pull request:
This adds an official `CupertinoPicker` example, this makes it much easier to try Cupertino picker.

### Preview

https://user-images.githubusercontent.com/48603081/141693637-5409f340-d2c2-44b9-a671-8a24690d0bac.mov

Related https://github.com/flutter/flutter/issues/72926

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

TahaTesser committed:
Add CupertinoPicker example

remove trailing space

test

Update example

fixes

Make example little colorful

Rename class

Kick tests

TahaTesser commented:
@gspencergoog 
Added test and cleaned up 
Thank you so much for the review!

gspencergoog reviewed (changes requested):
As with the other PRs, it would be good to add comments to the example code itself.

TahaTesser commented:
@gspencergoog 
Added comments, renamed class and provideed custom parameters 
Also I thought the example could use little color so I added emojis in the fruit strings

https://user-images.githubusercontent.com/48603081/152253863-5df7d3d2-01e3-473d-a3eb-2288915953e1.mov

gspencergoog reviewed (approved):
![32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d](https://user-images.githubusercontent.com/8867023/37737237-66e9c9dc-2d10-11e8-8797-13a405eace49.png)

Thanks, this looks great!

TahaTesser commented:
Hey @gspencergoog 
Can you please take a look https://master-api.flutter.dev/flutter/cupertino/CupertinoPicker-class.html?
Is it only master docs or this is how the emojis would be in the example. 
If emojis won't work then I could file a PR to remove those emojis.

gspencergoog commented:
Oooh, Yes, @TahaTesser I forgot that the web font that DartPad provides doesn't have Emojis.  I guess they'll just have to be removed.

gspencergoog commented:
It also looks like there's a bug in the picker that doesn't allow clicking on the item itself to set it, only clicking away from the chooser selects the item. This might only happen on web.

TahaTesser commented:
> It also looks like there's a bug in the picker that doesn't allow clicking on the item itself to set it, only clicking away from the chooser selects the item. This might only happen on web.

If I understand correctly, you're trying to tap on an item, expect to scroll to it and set it as the picked item.  This is actually `ListWheelScrollView` limitation, which I used by CupertinoPicker https://github.com/flutter/flutter/issues/38803

gspencergoog commented:
Huh, seems counterintuitive, but if that's intended behavior, then OK.
